packages/react-components/react-rating-preview/src/components/Rating/useRating.tsx

```diff
@@ -0,0 +1,100 @@
+import * as React from 'react';
+import {
+  getNativeElementProps,
+  isHTMLElement,
+  mergeCallbacks,
+  slot,
+  useControllableState,
+  useEventCallback,
+  useId,
+} from '@fluentui/react-utilities';
+import type { RatingProps, RatingState } from './Rating.types';
+import { Label } from '../../../../react-label/src/Label';
+import { RatingItem } from '../../RatingItem';
+
+/**
+ * Create the state required to render Rating.
+ *
+ * The returned state can be modified with hooks such as useRatingStyles_unstable,
+ * before being passed to renderRating_unstable.
+ *
+ * @param props - props from this instance of Rating
+ * @param ref - reference to root HTMLElement of Rating
+ */
+export const useRating_unstable = (props: RatingProps, ref: React.Ref<HTMLDivElement>): RatingState => {
+  const generatedName = useId('rating-');
+  const { name = generatedName, precision, readOnly, shape = 'star', size = 'medium', max = 5, onChange } = props;
+
+  const ratingId = useId('ratingLabel');
+  const countId = useId('countLabel');
+
+  const [value, setValue] = useControllableState({
+    state: props.value,
+    defaultState: props.defaultValue,
+    initialState: 0,
+  });
+
+  const [hoveredValue, setHoveredValue] = React.useState<number | undefined>(undefined);
+
+  const isRatingRadioItem = (target: EventTarget): target is HTMLInputElement =>
+    isHTMLElement(target, { constructorName: 'HTMLInputElement' }) && target.type === 'radio' && target.name === name;
```

Comment:
nit: maybe move `isRatingRadioItem` outside of useRating_unstable, to be a module-level function. You'd need add `name` as a parameter to the function, and pass it in when you call this function.